File: sdk/java/src/main/java/com/omnidex/integration/CreateChannelInput.java
package com.omnidex.integration;

import java.util.List;

public record CreateChannelInput(
    String id,
    String name,
    List<String> tags,
    String workspaceRoot,
    String dataSourceId
) {
    public CreateChannelInput {
        if (tags != null) tags = List.copyOf(tags);
    }
}

File: sdk/php/src/CreateChannelInput.php
<?php

declare(strict_types=1);

namespace Omnidex\Integration;

final readonly class CreateChannelInput
{
    /** @param list<string> $tags */
    public function __construct(
        public string $id,
        public string $name,
        public array $tags,
        public string $workspaceRoot,
        public string $dataSourceId,
    ) {
    }
}
